src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1415,13 +1405,11 @@ private MemberDeclarationSyntax ParseTypeDeclaration(SyntaxList<AttributeListSyn
             switch (this.CurrentToken.Kind)
             {
                 case SyntaxKind.ClassKeyword:
-                    return this.ParseClassOrStructOrInterfaceOrRoleDeclaration(attributes, modifiers);
-
                 case SyntaxKind.StructKeyword:
-                    return this.ParseClassOrStructOrInterfaceOrRoleDeclaration(attributes, modifiers);
-
                 case SyntaxKind.InterfaceKeyword:
-                    return this.ParseClassOrStructOrInterfaceOrRoleDeclaration(attributes, modifiers);
+                case SyntaxKind.ExplicitKeyword: // explicit extension
+                case SyntaxKind.ImplicitKeyword: // implicit extension
+                    return this.ParseClassOrStructOrInterfaceOrExtensionDeclaration(attributes, modifiers);
```

Comment:
did higher up logic already verify this is an extension and not an operator?

Reply:
Yes, calls to `ParseTypeDeclaration` are both preceded by attempts to parse operators